duecredit/collector.py

```diff
@@ -219,24 +227,23 @@ def cite(self, entry, **kwargs):
         # TODO: if cite is invoked but no path is provided -- we must figure it out
         # I guess from traceback, otherwise how would we know later to associate it
         # with modules???
+        path = kwargs.get('path', None)
+        if path is None:
+            raise ValueError('path must be provided')
+
         if isinstance(entry, DueCreditEntry):
             # new one -- add it
             self.add(entry)
-            entry_ = entry
+            entry_ = self._entries[entry.get_key()]
         else:
             entry_ = self._entries[entry]
-        entry_key = entry_.get_key()
 
-        # TODO: we must allow the same entry be present in multiple citations, so
-        # RF to do so
-        if entry_key not in self.citations:
-            self.citations[entry_key] = Citation(entry_, **kwargs)
-
-        citation = self.citations[entry_key]
+        entry_key = entry_.get_key()
+        if (path, entry_key) not in self.citations:
+           self.citations[(path, entry_key)] = Citation(entry_, **kwargs)
+        citation = self.citations[(path, entry_key)]
```

Comment:
dangerous leakage of internal "structure" of Citation... at least we might want add 

```
@staticmethod
def get_key(path, entry_key):
  return CitationKey(path, entry_key)
```
to the Citation, and then use it here, but also it feels that we should be able to
```
citation_key = Citation.get_key(path=path, entry_key=entry_key)
try:
  citation = self.citations[citation_key]
except KeyError:
  # create a new one, since unknown
  # don't we know path by now in **kwargs?
  self.citations[citation_key] = citation  = Citation(entry_, **kwargs)
  assert(citation.key == citation_key)
```